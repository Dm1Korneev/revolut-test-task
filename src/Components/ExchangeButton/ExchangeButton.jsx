import React from 'react';
import PropTypes from 'prop-types';

import { Button } from './ExchangeButton.styled';

const propTypes = {
  exchange: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
};

function ExchangeButton(props) {
  const {
    exchange, isActive,
  } = props;

  return (
    <Button type="button" onClick={exchange} disabled={!isActive}>
      Exchange
    </Button>
  );
}

ExchangeButton.propTypes = propTypes;

export default ExchangeButton;
