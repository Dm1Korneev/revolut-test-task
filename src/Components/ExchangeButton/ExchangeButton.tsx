import React, { FC } from 'react';

import { Button } from './ExchangeButton.styled';

type Props = {
  exchange: () => void;
  isActive: boolean;
}

const ExchangeButton: FC<Props> = (props): JSX.Element => {
  const {
    exchange, isActive,
  } = props;

  return (
    <Button type="button" onClick={exchange} disabled={!isActive}>
      Exchange
    </Button>
  );
};

export default ExchangeButton;
