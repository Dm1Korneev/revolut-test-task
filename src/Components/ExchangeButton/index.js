import commonHoc from 'Components/commonHoc';
import { exchange } from 'Redux/actions';
import { currenciesFromToSameSelector, exchagesValuesIsSetSelector } from 'Selectors/exchange';

import ExchangeButton from './ExchangeButton';

const mapStateToProps = (state) => ({
  isActive: exchagesValuesIsSetSelector(state) && !currenciesFromToSameSelector(state),
});

const mapDispatchToProps = {
  exchange,
};

export default commonHoc(ExchangeButton, {
  mapStateToProps,
  mapDispatchToProps,
});
