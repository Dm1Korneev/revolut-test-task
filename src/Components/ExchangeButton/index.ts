import commonHoc from 'Components/commonHoc';
import { exchange } from 'Redux/actions';
import { currenciesFromToSameSelector, exchagesValuesIsSetSelector } from 'Selectors/exchange';
import { StateType } from 'Redux/store';

import ExchangeButton from './ExchangeButton';

const mapStateToProps = (state: StateType): Record<string, any> => ({
  isActive: exchagesValuesIsSetSelector(state) && !currenciesFromToSameSelector(state),
});

const mapDispatchToProps = {
  exchange,
};

export default commonHoc(ExchangeButton, {
  mapStateToProps,
  mapDispatchToProps,
});
