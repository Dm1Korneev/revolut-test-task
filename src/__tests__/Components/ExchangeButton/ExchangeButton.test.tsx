import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ExchangeButton from 'Components/ExchangeButton/ExchangeButton';

describe('ExchangeButton component', () => {
  const exchange = jest.fn();
  const props = {
    exchange,
    isActive: true,
  };

  test('renders the component isActive = true', () => {
    const { asFragment } = render(<ExchangeButton {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders the component isActive = false', () => {
    const propslocal = { ...props, isActive: false };
    const { asFragment } = render(<ExchangeButton {...propslocal} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('change value', () => {
    const { getByRole } = render(<ExchangeButton {...props} />);
    fireEvent.click(getByRole('button'));
    expect(exchange).toHaveBeenCalledTimes(1);
  });
});
