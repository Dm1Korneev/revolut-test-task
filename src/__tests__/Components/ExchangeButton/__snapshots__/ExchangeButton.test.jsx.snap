// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeButton component renders the component isActive = false 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa zMcje"
    disabled=""
    type="button"
  >
    Exchange
  </button>
</DocumentFragment>
`;

exports[`ExchangeButton component renders the component isActive = true 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa zMcje"
    type="button"
  >
    Exchange
  </button>
</DocumentFragment>
`;
